common/credsregistry/fields.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/iancoleman/strcase"
+)
+
+// Fields is a grouping of constant values that dictate the keys that
+// can be present inside *creds.json file.
+var Fields = struct { // nolint:gochecknoglobals
+	Provider Field
+	// Tokens
+	AccessToken  Field
+	RefreshToken Field
+	Expiry       Field
+	ExpiryFormat Field
+	// Client ID, Secret
+	ClientId     Field
+	ClientSecret Field
+	// Basic Authentication
+	Username Field
+	Password Field
+	// Key
+	ApiKey Field
+	// Catalog variables
+	Workspace Field
+}{
+	Provider: Field{
+		Name:      "provider",
+		PathJSON:  "provider",
+		SuffixENV: "PROVIDER",
+	},
+	AccessToken: Field{
+		Name:      "accessToken",
+		PathJSON:  "accessToken",
+		SuffixENV: "ACCESS_TOKEN",
+	},
+	RefreshToken: Field{
+		Name:      "refreshToken",
+		PathJSON:  "refreshToken",
+		SuffixENV: "REFRESH_TOKEN",
+	},
+	Expiry: Field{
+		Name:      "expiry",
+		PathJSON:  "expiry",
+		SuffixENV: "EXPIRY",
+	},
+	ExpiryFormat: Field{
+		Name:      "expiryFormat",
+		PathJSON:  "expiryFormat",
+		SuffixENV: "EXPIRY_FORMAT",
+	},
+	ClientId: Field{
+		Name:      "clientId",
+		PathJSON:  "clientId",
+		SuffixENV: "CLIENT_ID",
+	},
+	ClientSecret: Field{
+		Name:      "clientSecret",
+		PathJSON:  "clientSecret",
+		SuffixENV: "CLIENT_SECRET",
+	},
+	Username: Field{
+		Name:      "username",
+		PathJSON:  "username",
+		SuffixENV: "USERNAME",
+	},
+	Password: Field{
+		Name:      "password",
+		PathJSON:  "password",
+		SuffixENV: "PASSWORD",
+	},
+	ApiKey: Field{
+		Name:      "apiKey",
+		PathJSON:  "apiKey",
+		SuffixENV: "API_KEY",
+	},
+	Workspace: Field{
+		Name:      "workspace",
+		PathJSON:  "substitutions.workspace",
+		SuffixENV: "WORKSPACE",
+	},
+}
+
+type Field struct {
+	Name      string
+	PathJSON  string
+	SuffixENV string
+}
+
+func (f Field) GetJSONReader(filepath string) *JSONReader {
+	return &JSONReader{
+		FilePath: filepath,
+		JSONPath: jsonPathTo(f.PathJSON),
+		CredKey:  f.Name,
+	}
+}
+
+func (f Field) GetENVReader(providerName string) *EnvReader {
+	return &EnvReader{
+		EnvName: envNameFor(providerName, f.SuffixENV),
+		CredKey: f.Name,
+	}
+}
+
+func getFields(info providers.ProviderInfo, withAccessToken bool) (handy.Lists[Field], error) {
+	lists := handy.Lists[Field]{}
+	requiredType := "required"
+	optionalType := "optional"
+
+	lists.Add(requiredType, Fields.Provider)
+
+	if withAccessToken {
+		lists.Add(requiredType, Fields.AccessToken)
+		lists.Add(optionalType, Fields.RefreshToken, Fields.Expiry, Fields.ExpiryFormat)
+	}
+
+	switch info.AuthType {
+	case providers.ApiKey:
+		lists.Add(requiredType, Fields.ApiKey)
+	case providers.Basic:
+		lists.Add(requiredType, Fields.Username, Fields.Password)
+	case providers.None:
+	case providers.Oauth2:
+		lists.Add(requiredType, Fields.ClientId, Fields.ClientSecret)
+	default:
+		return nil, ErrProviderInfo
+	}
+
+	options := info.Oauth2Opts
+	if options != nil {
+		// TODO workspace can happen not only to Oauth2
```

Comment:
As of now we can conclude that workspace is needed for Oauth2. Other auth types dont have this flag. Here is a PR for changing models: https://github.com/amp-labs/openapi/pull/15